backend/routes.py

```
@@ -20,87 +20,73 @@
         # Check if user is admin or Materials department
-        if not (session.get('is_admin', False) or session.get('department') == 'Materials'):
-            log_security_event('insufficient_permissions', f'Materials management access denied for user {session.get("user_id")}')
-            return jsonify({'error': 'Materials management privileges required'}), 403
+        if user_payload.get('is_admin', False) or user_payload.get('department') == 'Materials':
+            return f(*args, **kwargs)
 
-        return f(*args, **kwargs)
+        return jsonify({'error': 'Materials management privileges required', 'code': 'MATERIALS_MANAGER_REQUIRED'}), 403
     return decorated_function
 
 def register_routes(app):
```

Pylint (3.3.7)

[refactor] Too many local variables (54/15)

(R0914)